File: java/Vulnerable2.java
/**
 * Additional vulnerable Java code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

import java.sql.*;
import java.io.*;

public class Vulnerable2 {
    
    // SQL Injection
    public ResultSet searchUsers(String searchTerm) throws SQLException {
        String query = "SELECT * FROM users WHERE name LIKE '%" + searchTerm + "%'";
        Statement stmt = connection.createStatement();
        return stmt.executeQuery(query);
    }
    
    // Hardcoded secrets
    private static final String DB_PASSWORD = "admin123";
    private static final String API_KEY = "sk_live_51H3ll0W0rld123456789";
    
    // Command Injection
    public void runCommand(String command) throws IOException {
        Runtime.getRuntime().exec(command);
    }
    
    // Path Traversal
    public String readConfigFile(String filename) throws IOException {
        return new String(Files.readAllBytes(Paths.get("/etc/config/" + filename)));
    }
}

File: kotlin/Vulnerable2.kt
/**
 * Additional vulnerable Kotlin code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

import java.sql.Statement

// SQL Injection
fun searchUsers(searchTerm: String): ResultSet {
    val query = "SELECT * FROM users WHERE name LIKE '%$searchTerm%'"
    val stmt: Statement = connection.createStatement()
    return stmt.executeQuery(query)
}

// Hardcoded secrets
const val DB_PASSWORD = "admin123"
const val API_KEY = "sk_live_51H3ll0W0rld123456789"

// Command Injection
fun runCommand(command: String) {
    Runtime.getRuntime().exec(command)
}

// Path Traversal
fun readConfigFile(filename: String): String {
    return java.io.File("/etc/config/$filename").readText()
}

